lib/iris/cube.py

```diff
@@ -3650,6 +3650,23 @@ def __itruediv__(self, other):
     __pow__ = iris.analysis.maths.exponentiate
     # END OPERATOR OVERLOADS
 
+    def _trim_ancils(self, dims_to_collapse):
```

Comment:
Frankly I'm not keen on the implicit use of 'self.cube', and the fact this is now the only way in which that will be accessed.  Maybe just a more transparent naming can fix that, and/or a docstring/comment of purpose/usage.
Best I can think of on the fly would be something like..
```
    def _trimmed_cube(self, dims_to_collapse):
        # Get a copy of self.cube, minus any ancillary-variables or cell-measures that map the collapse dimensions.
```

Reply:
**Alternatively**
If we are going to be using, and maybe copying this operation elsewhere, then make it standalone general function that accepts the original cube as an argument, e.g. 
```
def _remove_ancils_and_cms_mapping_dims(cube, dimensions_affected):
    """
    Remove ancillary variables and cell measures that map to specific dimensions.

    To be used by operations that modify or remove dimensions.
    Note: does nothing to (aux)-coordinates.  We expect those will be handled explicitly in the calling operation.

    """